lib/httpclient.rb

```diff
@@ -1073,8 +1077,10 @@ def load_environment
       # HTTP_* is used for HTTP header information.  Unlike open-uri, we
       # simply ignore http_proxy in CGI env and use cgi_http_proxy instead.
       self.proxy = getenv('cgi_http_proxy')
-    else
+    elsif ENV.key?('http_proxy')
```

Comment:
Would suggest changing this line to `elsif ENV.key?getenv('http_proxy')` to deal with my comment above